### PR TITLE
WE-6492 Add social navigation to sidebar

### PR DESCRIPTION
**Pull request by walsh9**
@noslouch @marineb 
Trying a few different things with the SASS/CSS here. Please let me know what you like and don't like.
1. Giving BEM conventions a try as suggested by @marineb 
2. Trying out explicitly importing mixins where they are used, instead of just having them all at the top of app.scss. (mixins/vars files should generate no css on their own)

This change passes the showPlayer state into the site-chrome, so it can change layout based on the player open/close state.

This change also assumes the settings link will be removed somewhere else.

### CODE REVIEW

**Comment by walsh9** [comment]
https://jira.wnyc.org/browse/WE-6492

File: app/components/site-chrome/template.hbs [hunk 1]
```diff
@@ -148,6 +148,9 @@
     <!-- .x-scrollable-body -->
 
     {{#if media.isLargeAndUp}}
+    <div class="sitechrome__nav-footer">
```
Comment (noslouch) [inline app/components/site-chrome/template.hbs]: why is this `sitechrome` and not `site-chrome`? I'm thinking of how `social-icon-list` is hyphenated.

File: app/styles/mixins/_lists.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@mixin list-base {
+    list-style: none;
+    padding: 0;
```
Comment (noslouch) [inline app/styles/mixins/_lists.scss]: 4 spaces here instead of 2

**Comment by noslouch** [comment]
ok, I think I like the BEM stuff, but what is the difference between `__` and `--`? I think the hyphens mean something like a state and the underscores mean something like a nested element?

also very much on board with the idea of importing mixins that do not generate CSS. let's keep it up!

BUT, here's the bad news:
I see this is also [in qxr.](https://github.com/nypublicradio/wqxr-web-client/commit/6d1c53f2cf4e44f986e794557a16c0622b4fc62d) So that means this component should go into nypr-ui. _Additionally_, it needs to allow for a list of links to be passed in so that wnyc-web-client vs wqxr-web-client can specify their own social urls.

File: app/styles/mixins/_layout.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@mixin l-bottom {
```
Comment (marineb) [inline app/styles/mixins/_layout.scss]: Shouldn't it be `l_bottom`?

**Comment by marineb** [comment]
Aweseome @walsh9 !!

File: app/components/site-chrome/template.hbs (hunk 1 above)
Comment (walsh9) [inline app/components/site-chrome/template.hbs]: didn't want to rename the rest of the sitechrome classes yet. Was mostly focused on the new component. Using `site-chrome` makes sense though.

File: app/styles/mixins/_layout.scss (hunk 3 above)
Comment (walsh9) [inline app/styles/mixins/_layout.scss]: not sure. I couldn't find any BEM official guidance for prefixing stuff like this.  I think it's up to us however we want to define our prefixes.

Comment (marineb) [inline app/styles/mixins/_layout.scss]: Actually, you're right. And we could follow this for instance? https://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/

Comment (marineb) [inline app/styles/mixins/_layout.scss]: It's kind of nice to be able to know from looking at the CSS that something is a component. cc @noslouch

Comment (noslouch) [inline app/styles/mixins/_layout.scss]: we just need to define what's what. I'm still unsure what the difference is between `--` and `__`.

Comment (marineb) [inline app/styles/mixins/_layout.scss]: @noslouch the `__` is to define an Element that belongs to a Block. And the `--` is to define a Modifier (to an element or a block). 

So for instance: `.alert__header--success`, or `.alert--success`.

Comment (noslouch) [inline app/styles/mixins/_layout.scss]: So in this example, there could be an alert__header and an alert__header--success or an alert and alert--success?

Comment (marineb) [inline app/styles/mixins/_layout.scss]: Yes, all of those are valid!

File: app/components/site-chrome/template.hbs [hunk 4]
```diff
@@ -139,6 +139,30 @@
...
+      {{social-icon-list class="social-icon-list"}}
+    </div>
+    <div class="l-bottom">
+        <div class="box box--nopad">
+
+
+          <ul class="list list--noborder flush alignright">
+            {{!-- <div class="h5 text--heavy alignright"> --}}
+            {{#if features.autoplayPrefs}}
```
Comment (noslouch) [inline app/components/site-chrome/template.hbs]: wasn't this moved to the user-settings component?

**Pull request merged**